functions/getShippingRatesFromShipEngine/main.js

```diff
@@ -0,0 +1,22 @@
+const functions = require('firebase-functions');
+const calculateShippingRates = require('./services/calculateShippingRates');
+const getConfig = require('./services/getConfig');
+
+const getShippingRatesFromShipEngine = async (data) => {
+    const { shipment, shipmentId } = data;
+    if (!!shipment && !!shipmentId) {
+        throw new functions.https.HttpsError(
+            'failed_precondition',
```

Comment:
it is failed-precondition and also here I think this is a bad request which maps to invalid-argument